lib/content-store/rest-handler.js

```diff
@@ -66,42 +66,49 @@ function transform(obj) {
   }, {});
 }
 
-const RestHandler = {
-  name: 'HTTP REST',
-  get(pathname) {
-    return new Promise((resolve, reject) => {
-      if (!BASEURL) {
-        return reject(new Error('Content store base URL not set'));
-      }
-
-      return request({
-        method: 'GET',
-        json: true,
-        timeout: TIMEOUT,
-        uri: `${BASEURL}/api/pages/with-path/${pathname}/`,
-        headers: {
-          Authorization: `Bearer ${AUTH_TOKEN}`,
-        },
-      }, (error, response, body) => {
-        if (error) {
-          return reject(error);
-        }
+function fetchData(endpoint, pathname) {
+  return new Promise((resolve, reject) => {
+    if (!BASE_URL) {
+      return reject(new Error('Content store base URL not set'));
+    }
 
-        if (response.statusCode === 200) {
-          const record = body;
-          let layout = 'content-simple';
+    return request({
+      method: 'GET',
+      json: true,
+      timeout: TIMEOUT,
+      uri: `${BASE_URL}/api/${endpoint}/${pathname}`,
+      headers: {
+        Authorization: `Bearer ${AUTH_TOKEN}`,
+      },
+    }, (error, response, body) => {
+      if (error) {
+        return reject(error);
+      }
 
-          if (record.sidebarOrder && record.sidebar.length && record.sidebar.length > 0) {
-            layout = `content-sidebar-${record.sidebarOrder}`;
-          }
+      if (response.statusCode === 200) {
+        const record = body;
+        let layout = 'content-simple';
 
-          record.layout = layout;
-          return resolve(transform(record));
+        if (record.sidebarOrder && record.sidebar.length && record.sidebar.length > 0) {
+          layout = `content-sidebar-${record.sidebarOrder}`;
         }
 
-        return reject(new Error(response));
-      });
+        record.layout = layout;
+        return resolve(transform(record));
+      }
+
+      return reject(new Error(body.message));
     });
+  });
+}
+
+const RestHandler = {
+  name: 'HTTP REST',
+  get(pathname) {
+    return fetchData('pages/with-path', `${pathname}/`); // include trailing slash. It's required by Wagtail API routes
+  },
+  preview(pathname) {
+    return fetchData('preview-pages', pathname);
```

Comment:
Nope, because it's followed by query string. I do think Wagtail routes should have the trailing slash optional though.